core/templates/dev/head/collection_player/CollectionPlayer.js

```diff
@@ -0,0 +1,121 @@
+// Copyright 2015 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controller for the learner's view of a collection.
+ *
+ * @author henning.benmax@gmail.com (Ben Henning)
+ */
+
+oppia.constant(
+  'COLLECTION_DATA_URL', '/collectionhandler/data/<collection_id>');
+
```

Comment:
I spent a good bit debugging after your comment. I'm not sure why the extra div is there (this is apparently common for jquery to do). That's probably what the problem is. I was able to get a very nice and smooth transition just using CSS transitions, but it had the unfortunate side effect of requiring an explicit max-height value which is not known at runtime. This turned out to be a bit more complicated for me to solve than I originally thought it was going to be, so I'm going to file an issue for it instead of fixing it in this commit.

Thanks, Amit!